woocommerce/woocommerce-android
# [Build] Upgrade Gradle to 8.2.1 and AGP to 8.1.0

### Description


Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

This PR upgraded Gradle to [8.2.1](https://docs.gradle.org/8.2.1/release-notes.html) and AGP to [8.1.0](https://developer.android.com/build/releases/gradle-plugin#8-1-0).

FYI: The [8.1.1](https://docs.gradle.org/8.1.1/release-notes.html) Gradle upgrades had to happen in 2 parts (87b30ea64066817ffb76fc3378f7fb8d12c81581 and d6e0d5b944be4eb1be9613e81c0b18fe4b18206b). This is because of the breaking changes that this upgrade introduces, including the `AGP` upgrade and its associated breaking changes (namespace, build option, etc).

-----

🚫 Blocked By:


<details>
    <summary>CI Failure</summary>

```
[2023-07-17T11:47:46Z] FAILURE: Build failed with an exception.
--
  | [2023-07-17T11:47:46Z]
  | [2023-07-17T11:47:46Z] * Where:
  | [2023-07-17T11:47:46Z] Build file '/var/lib/buildkite-agent/builds/ci-android-i-03301d03eb1c8e4e5-1/automattic/woocommerce-android/quicklogin/build.gradle' line: 2
  | [2023-07-17T11:47:46Z]
  | [2023-07-17T11:47:46Z] * What went wrong:
  | [2023-07-17T11:47:46Z] An exception occurred applying plugin request [id: 'com.android.test']
  | [2023-07-17T11:47:46Z] > Failed to apply plugin 'com.android.internal.test'.
  | [2023-07-17T11:47:46Z]    > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
  | [2023-07-17T11:47:46Z]       Your current JDK is located in /usr/lib/jvm/java-11-openjdk-11.0.19.0.7-1.amzn2.0.1.x86_64
  | [2023-07-17T11:47:46Z]       You can try some of the following options:
  | [2023-07-17T11:47:46Z]        - changing the IDE settings.
  | [2023-07-17T11:47:46Z]        - changing the JAVA_HOME environment variable.
  | [2023-07-17T11:47:46Z]        - changing `org.gradle.java.home` in `gradle.properties`.
```

</details>

-----

Breaking Change List:

1. [Enable build config build feature for quicklogin and woocommerce](https://github.com/woocommerce/woocommerce-android/commit/db2340bb8a5fa688f29cc35b61f8d5d067192cbe)
2. [Update google services plugin to 4.3.15](https://github.com/woocommerce/woocommerce-android/commit/5966c09c3907bd625e3bd12c582711ce6aebf319)
3. [Resolve all violating non final res references for main](https://github.com/woocommerce/woocommerce-android/commit/2e77a0c9ee5f6e97699bbc53f04c519591b7e5eb)
4. [Resolve all violating non final res references for android test](https://github.com/woocommerce/woocommerce-android/commit/c7147dd2618cf10f5a0fa5dae429818007106925)
5. [Replace dependency substitution substitute with method with using](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/d9ed4bb9b982ad979cd4ce63731e2a0f43fc02e3)

Test Resolution List:

1. [Fix gson related test failures on site picker view model test](https://github.com/woocommerce/woocommerce-android/commit/9b3cf55e00d60b3fc14fcab07e9967abf5a058dd)

Lint Suppression List:

1. [Suppress new api lint warning for context ext](https://github.com/woocommerce/woocommerce-android/commit/86e64288484ef27892771a86eb1cb5f677207bd1)

Dependency Versions Refactor List:

1. [Extract google services plugin related version](https://github.com/woocommerce/woocommerce-android/commit/f2c07a94c644c1c1720517da25b4a4d2a1dd2be3)
2. [Extract wzieba tracks plugin related version](https://github.com/woocommerce/woocommerce-android/commit/204a619363973ce4402472c3faaede68f4482f2f)
6. [Extract sentry plugin related version](https://github.com/woocommerce/woocommerce-android/commit/ef4e67592972b16e8b871dec6829d674a911b818)
7. [Extract violation comments plugin related version](https://github.com/woocommerce/woocommerce-android/commit/b71e2222424f2628a2d7f68afc0b46fd59b230a0)

-----

## Associated libraries

You can reference the `[Build] Upgrade Gradle to 8.1.1 and AGP to 8.0.2` PRs of the below associated libraries for testing/verification purposes:

- [x] [MediaPicker#69](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/69)
- [x] [Login#119](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/119)
- [x] [FluxC#2772](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2772)

-----

### Testing instructions


1. Verify that all the CI checks are successful.
2. Smoke test the app, without composite builds.
3. Smoke test the app again, but this time, using the composite builds for all 3 libraries (following the instructions in [local-builds.gradle-example](https://github.com/woocommerce/woocommerce-android/blob/trunk/local-builds.gradle-example) file). PS: Make sure you're on the `Gradle & AGP` branches on all 3 libraries.

-----

### Testing instructions

- [x] Update all 3 library references to PR hash (this is done for testing purposes):
  - [x] [mediapickerVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/ee964891667eaa30bf774c0225f20980779e9832) (v1) + [mediapickerVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/75bc96949c64214a7c62f2236a243903e7eabba9) (v2)
  - [x] [wordPressLoginVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/d692875c38b7978f94cc1058bd09d0a1cc767683) (v1) + [wordPressLoginVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/d7317feb17a5641aa435dd60d0c2cf5b692b8cbe) (v2)
  - [x] [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/7aee36cd457296215fec3eda240b5dd29af54275) (v1) + [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/732f43d9c1892b29add62bf11fd44cfae28dda44) (v2) + [fluxCVersion](https://github.com/woocommerce/woocommerce-android/pull/9427/commits/9ea0b97db6d7a452df250bb753a4fc5c4a1cc6ab) (v3)
- [x] Merge all 3 library PRs into their `trunk` (after them being code reviewed and approved)
  - [x] [MediaPicker#69](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/69)
  - [x] [Login#119](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/119)
  - [x] [FluxC#2772](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2772)
- [x] Update all 3 library references to `trunk` hash or tag:
  - [x] [mediapickerVersion](TODO)
  - [x] [wordPressLoginVersion](TODO)
  - [x] [fluxCVersion](TODO)
- [x] Remove `status: do not merge` label.

-----
